src/blister.h

```diff
@@ -352,6 +355,147 @@ extern void ConvBlist (
+    else
+        *to |= ((from >> -shift) & m);
+}
+
+
+static inline  __attribute__((always_inline)) void CopyBits(const UInt * fromblock,
+              UInt         frombit,
+              UInt *       toblock,
+              UInt         tobit,
+              UInt         nbits)
+{
+    UInt tailbits;
+    UInt x;
+    UInt wholeblocks;
+    if (!nbits)
+        return;
+    /* Normalize */
+    fromblock += frombit / BIPEB;
+    frombit %= BIPEB;
+    toblock += tobit / BIPEB;
+    tobit %= BIPEB;
+    /* If the alignment of the two data blocks matches, things are relatively
+     * easy
+     */
+    if (frombit == tobit) {
+        /* if the first and last words are the same word */
+        if ((frombit + nbits) < BIPEB) {
+            CopyInWord(toblock, frombit, frombit + nbits - 1, *fromblock, 0);
+            return;
+        }
+        /* do we need to start by copying a partial word */
+        if (frombit) {
+            CopyInWord(toblock, frombit, BIPEB - 1, *fromblock, 0);
+            fromblock++;
+            toblock++;
+            nbits -= (BIPEB - frombit);
+            frombit = 0;
+            tobit = 0;
+        }
+        /* Now move whole words */
+        if ((wholeblocks = nbits / BIPEB))
+            memcpy((void *)toblock, (void *)fromblock,
+                   sizeof(UInt) * wholeblocks);
+        toblock += wholeblocks;
+        fromblock += wholeblocks;
+        nbits %= BIPEB;
+        /* Finally, we may need to finish with another partial word */
+        if (nbits)
+            CopyInWord(toblock, 0, nbits - 1, *fromblock, 0);
+        return;
+    }
+
+    /* Otherwise the bits are not aligned and we will be shifting */
+
+    if (tobit) {
+        /* How many bits are we going to put into the first destination word
+         */
+        if (tobit + nbits <= BIPEB)
+            tailbits = nbits;
+        else
+            tailbits = BIPEB - tobit;
+        /* We might be able to get all we need from the first source word */
+        if (frombit + tailbits <= BIPEB) {
+            CopyInWord(toblock, frombit, frombit + tailbits - 1, *fromblock,
+                       (tobit - frombit));
+            frombit += tailbits;
+        }
+        else {
+            CopyInWord(toblock, frombit, BIPEB - 1, *fromblock,
+                       (tobit - frombit));
+            fromblock++;
+            CopyInWord(toblock, 0, tailbits + frombit - 1 - BIPEB,
+                       fromblock[1], tobit + BIPEB - frombit);
+            frombit += tailbits - BIPEB;
+        }
+        toblock++;
+        nbits -= tailbits;
+        tobit = 0;
+    }
+    /* Main loop for long copies fills whole blocks of destination */
+    UInt m1 = mask(frombit, BIPEB - 1);
+    while (nbits >= BIPEB) {
+        x = (*fromblock++ & m1) >> frombit;
+        x |= (*fromblock & ~m1) << (BIPEB - frombit);
+        *toblock++ = x;
+        nbits -= BIPEB;
+    }
+    /* Finally we may need to fill up a partial block at destination */
```

Comment:
Why is that comment attached to the preceding loop, instead of the following `if` (to which it refers, doesn't it?)

I.e.: maybe put the empty line before the comment, not after it?